lib/frontend/terminal_view.dart

```diff
@@ -241,21 +238,44 @@ class _TerminalViewState extends State<TerminalView> {
       },
       child: Container(
         constraints: BoxConstraints.expand(),
-        child: CustomPaint(
-          painter: TerminalPainter(
-            terminal: widget.terminal,
-            view: widget,
-            oscillator: oscillator,
-            focused: focused,
-            charSize: _cellSize,
-          ),
+        child: Stack(
```

Comment:
https://dartpad.dev/31519700f255ff2daccb207224fbe18c?null_safety=true

It seems that flutter `Stack` always force repaint all of its children when one of its children needs to be repainted. 
Similarly, once `CursorView` is repainted, `TerminalPainter.paint` is always called. Thus I think there might be no need to split the Cursor into a separate Widget.

Reply:
I tried it with the draw visualization option in the debugger settings and there only the cursor did repaint. But I didn't check if the paint of the CustomPainter was actually not called.

Reply:
"Repaint Rainbow"
![CleanShot 2021-04-18 at 17 32 13](https://user-images.githubusercontent.com/6693130/115151198-1ee26e00-a06c-11eb-92ba-e3f83af4e729.gif)

Reply:
So the paint method is called each time (so your assumption regarding the Stack is correct).
Nevertheless there seems to be a difference as the rainbow doesn't cycle to its colors. Not sure where this difference actually happens and what it means in terms of performance.
Needs more investigation I think

Reply:
https://dartpad.dev/22078b0cf20196256adfa34a553f8cf1
Here a modified version of the sample that shows that build is only called for the affected widget but paint is called for both of them.
I used Texts with a gesture detector as the TextButton was causing a whole bunch of redraws by itself

Reply:
https://stackoverflow.com/questions/46702376/how-to-ensure-my-custompaint-widget-painting-is-stored-in-the-raster-cache/46706868#46706868

This might help.

Reply:
Really interesting find!

Reply:
I tried to wrap TerminalPainter with RepaintBoundary, it works but performance seems to have regression. When running cmatrix -r, CPU utilization is about 10% higher.

Reply:
That is interesting... does it make a difference in Debug/Release?